lib/ansible/modules/storage/glusterfs/gluster_heal_facts.py

```diff
@@ -0,0 +1,166 @@
+    choices: ["self-heal", "rebalance"]
+    description:
+      - Determines which facts are to be returned. If the status_filter is
+        self-heal, status of self-heal, along with the number of files still
+        in process are returned. If status_filter is rebalance,
+        rebalance status is returned.
+requirements:
+  - GlusterFS > 3.2
+'''
+
+EXAMPLES = '''
+# Gather facts about all gluster hosts in the cluster and return self-heal
+# status along with the number of files still in process:
+- gluster_heal_facts:
+    name: test_volume
+    status_filter: self-heal
+- debug:
+    var: gluster_hosts
+'''
+
+RETURN = '''
+name:
+    description: GlusterFS volume name
+    returned: always
+    type: string
+status_filter:
+    description: Whether self-heal or rebalance status is to be returned
+    returned: always
+    type: string
+heal_info:
+    description: List of files that still need healing process
+    returned: On success
+    type: string
+rebalance_status:
+    description: Status of rebalance operation
+    returned: On success
+    type: string
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from distutils.version import LooseVersion
+
+glusterbin = ''
+
+
+def run_gluster(gargs, **kwargs):
+    global glusterbin
+    global module
+    args = [glusterbin, '--mode=script']
+    args.extend(gargs)
+    try:
+        rc, out, err = module.run_command(args, **kwargs)
+        if rc != 0:
+            module.fail_json(msg='error running gluster (%s) command (rc=%d): %s' %
+                             (' '.join(args), rc, out or err), exception=traceback.format_exc())
+    except Exception as e:
+        module.fail_json(msg='error running gluster (%s) command: %s' % (' '.join(args),
+                                                                         to_native(e)), exception=traceback.format_exc())
+    return out
+
+
+def get_self_heal_status(name):
+    out = run_gluster(['volume', 'heal', name, 'info'])
+    raw_out = out.split("\n")
+    heal_info = []
+    # return files that still need healing.
+    for line in raw_out:
+        if 'Brick' in line:
+            br_dict = {}
+            br_dict['brick'] = line.strip().strip("Brick")
+        elif 'Status' in line:
+            br_dict['status'] = line.split(":")[1].strip()
+        elif 'Number' in line:
+            br_dict['no_of_entries'] = line.split(":")[1].strip()
+            if int(br_dict['no_of_entries']) != 0:
+                heal_info.append(br_dict)
+            else:
+                continue
+    return heal_info
+
+
+def get_rebalance_status(name):
+    out = run_gluster(['volume', 'rebalance', name, 'status'])
+    return out
+
+
+def main():
+    global module
+    argument_spec = dict(name=dict(type='str', required=True, aliases=['volume']),
+                         status_filter=dict(default='self-heal', choices=['self-heal', 'rebalance']),
```

Comment:
Can you make the parameters one per line?

See: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/storage/glusterfs/gluster_volume.py#L447